service/src/test/org/apache/hive/service/cli/TestRetryingThriftCLIServiceClient.java

```diff
@@ -66,13 +69,30 @@ public void init() {
         "org.apache.hadoop.hive.ql.security.authorization.plugin.sqlstd.SQLStdHiveAuthorizerFactory");
   }
 
-  private void startHiveServer() throws InterruptedException {
-    // Start hive server2
-    server = new HiveServer2();
-    server.init(hiveConf);
-    server.start();
-    Thread.sleep(5000);
-    System.out.println("## HiveServer started");
+  private void startHiveServer() throws Exception {
```

Comment:
that looks similar to TestHS2HttpServer#beforeTests, would be nice to extract common code into a base class

Reply:
@deniskuzZ I have made a few changes please have a look!